src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -93,35 +101,50 @@ func (client *Client) targetQOS(req *http.Request) int {
 // pipeData pipes data between the client and proxy connections.  It's also
 // responsible for responding to the initial CONNECT request with a 200 OK.
 func pipeData(clientConn net.Conn, connOut net.Conn, req *http.Request) {
-
-	var wg sync.WaitGroup
-	wg.Add(1)
-	// Start piping from client to proxy
-	go func() {
-		io.Copy(connOut, clientConn)
-		// Force closing if EOF at the request half or error encountered.
-		// A bit arbitrary, but it's rather rare now to use half closing
-		// as a way to notify server. Most application closes both connections
-		// after completed send / receive so that won't cause problem.
-		wg.Wait()
-		clientConn.Close()
-		connOut.Close()
-	}()
-
 	// Respond OK
 	err := respondOK(clientConn, req)
```

Comment:
Just doing this at the beginning is far better -- don't know why I didn't think of that thanks.